client/app/dashboard/backlog.js

```diff
@@ -1,58 +1,88 @@
 angular.module('hunt.backlog', [])
 
-.controller('BacklogController', function ($scope, $rootScope, $location, $window, Backlog) {
+.controller('BacklogController', function ($scope, $rootScope, $location, $window, Backlog, AppSubmit) {
   $rootScope.backlogs = [];
+
   $rootScope.selectedBacklogIndex;
 
+  // Function that retrieves all backlogs for given user
   $scope.getBacklogs = function () {
-    // Query the DB for all backlogs using the function in server controller 
-    //  On success, assign $scope.backlogs to the data returned from query
     Backlog.getBacklogs($window.localStorage.getItem('user_id'))
       .then(function (data) {
         $rootScope.backlogs = data;
       })
       .catch(function (error) {
-        console.log('Error initializing backlogs: ', error);
+        console.log('Error getting backlogs: ', error);
       });
   };
 
+  // Function that removes backlog
   $scope.removeBacklog = function () {
 
   };
 
+  // Function that edits an existing backlog
+  $scope.editBacklog = function (backlog) {
+    Backlog.editBacklog(backlog)
+        .then(function (backlog) {
+          $rootScope.backlogs.splice($rootScope.selectedBacklogIndex, 1, backlog);
+        })
+        .catch(function (error) {
+          console.log("Error editing backlog: ", error);
+        });
+  };
+
+  // Function that moves backlog to application submitted state
   $scope.moveToAppSubmitted = function () {
+    var backlogChanges = {
+        user_id: $window.localStorage.getItem('user_id'),
+        application_id: $rootScope.selectedApplicationId,
+        id: $rootScope.backlogID,
+        notes: $scope.backlogNotes,
+        status: 'Pending'
+      };
 
+      AppSubmit.addAppSubmit(backlogChanges)
+        .then(function (backlog) {
+          $rootScope.appSubmits.push(backlog);
+          $rootScope.backlogs.splice($rootScope.selectedBacklogIndex, 1, backlog);
```

Comment:
We'd want to push the new appSubmit onto $rootScope.appSubmits, and the backlog onto $rootScope.backlogs. I think it would be better to separate out the logic of doing the backlog change from adding the new appSubmit--right now both are living in this function which is kind of hard to follow.